modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/context/StoreContext.js

```diff
@@ -9,67 +9,47 @@
  * distribution rights of the Software.
  */
 
-import React, {createContext, useCallback, useContext, useReducer} from 'react';
+import React, {createContext, useContext, useReducer} from 'react';
+
+import {ADD_WARNING} from '../actions/types';
 
 const INITIAL_STATE = {
-	historicalWarning: false,
 	publishedToday: false,
 	warning: false,
 };
-const ADD_HISTORICAL_WARNING = 'add-historical-warning';
-const ADD_WARNING = 'add-warning';
 
 const noop = () => {};
 
-export const StoreContext = createContext([INITIAL_STATE, noop]);
+export const StoreDispatchContext = React.createContext(() => {});
+export const StoreStateContext = createContext([INITIAL_STATE, noop]);
 
 function reducer(state = INITIAL_STATE, action) {
-	if (action.type === ADD_HISTORICAL_WARNING) {
-		return state.historicalWarning
-			? state
-			: {...state, historicalWarning: true};
-	}
-	else if (action.type === ADD_WARNING) {
-		return state.warning ? state : {...state, warning: true};
+	let nextState = state;
+
+	switch (action.type) {
+		case ADD_WARNING:
+			nextState = state.warning ? state : {...state, warning: true};
+			break;
+		default:
+			return state;
 	}
 
-	return state;
+	return nextState;
 }
 
 export function StoreContextProvider({children, value}) {
-	const stateAndDispatch = useReducer(reducer, {...INITIAL_STATE, ...value});
+	const [state, dispatch] = useReducer(reducer, {...INITIAL_STATE, ...value});
 
 	return (
-		<StoreContext.Provider value={stateAndDispatch}>
-			{children}
-		</StoreContext.Provider>
+		<StoreDispatchContext.Provider value={dispatch}>
+			<StoreStateContext.Provider value={state}>
+				{children}
+			</StoreStateContext.Provider>
+		</StoreDispatchContext.Provider>
 	);
 }
 
-export function useHistoricalWarning() {
-	const [state, dispatch] = useContext(StoreContext);
-
-	const addHistoricalWarning = useCallback(() => {
-		dispatch({
-			type: ADD_HISTORICAL_WARNING,
-		});
-	}, [dispatch]);
-
-	const hasHistoricalWarning = state.historicalWarning;
-
-	return [hasHistoricalWarning, addHistoricalWarning];
-}
-
-export function useWarning() {
-	const [state, dispatch] = useContext(StoreContext);
-
-	const addWarning = useCallback(() => {
-		dispatch({
-			type: ADD_WARNING,
-		});
-	}, [dispatch]);
-
-	const hasWarning = state.warning;
-
-	return [hasWarning, addWarning];
-}
+/**
+ * @see https://react-redux.js.org/api/hooks#usedispatch
+ */
+export const useDispatch = () => useContext(StoreDispatchContext);
```

Comment:
I understand you're following the Redux pattern here, and we also have a `useDispatch` hook in `layout-content-page-editor-web`, but:

1. You're not actually using Redux.
2. This hook literally does nothing other than call another hook.

Which means that you could just call the other hook, and avoid an unnecessary layer of indirection. FWIW, see [this commit in `layout-content-page-editor-web` to compare what it used to look like before](https://github.com/liferay/liferay-portal/commit/2f2d2f67ccbb9803b2b5cd36760f138bcb253b8d).

But this is non-blocking feedback...

![just-sayin](https://user-images.githubusercontent.com/7074/111175303-95d9a200-85a8-11eb-9356-3a6895f9fca3.gif)

In general I'd be wary of bringing in too much Redux-like boilerplate before you really need it (like the separate files for the actions and the action types, when you currently only have one action, at least as of this PR). One of the main complaints people have about Redux is [the boilerplate](https://redux.js.org/recipes/reducing-boilerplate).